evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -1861,6 +1933,9 @@ where
             anyhow::Error::msg("Invalid conversion when setting public values targets.")
         })?;
 
+        // TODO(sdeng): Set to false when this segment contains no Keccak operations.
+        root_inputs.set_bool_target(self.root.use_keccak_tables, true);
```

Comment:
I think you forgot to address this

Reply:
This is addressed in #657

Reply:
I think we can group the two PRs, they aren't big and the addition here doesn't really make sense without actually leveraging the new feature IMO

Reply:
You will see some TODOs, as I prefer not to include everything in a single PR. This PR separates the security-affected changes from the non-security-affected ones.

Reply:
> This PR separates the security-affected changes from the non-security-affected ones.

I'm not sure what you mean by this. Modifying even a tiny piece of the logic (what the follow up PR does) could break security.

What I meant as a whole is that despite reviews, we won't merge this into `develop` without extensive tests and benchmarking, which are possible only once both distinct PRs are merged together, and if changes are needed post testing, then this will have to be done on the whole scope of change, hence my initial comment. Unless PRs are not hundreds of lines long / touching different pieces of logic, I think it's usually best to group incremental pieces together (it also gives reviewers a change to see the whole logic flow without missing bits). But this being said, feel free to keep this is separate if you prefer.

Reply:
Sometimes we may need a cryptographer to review certain (math related) PRs, and there’s no need to send over large PRs for review. When a PR is too big, there’s a higher chance that the reviewer might overlook important details. For example, my other PR only involves refactoring and setting public inputs for the root circuit. It won’t compromise security like introducing bugs in the verifier/circuits could.

Reply:
Hi Linda, thanks for reviewing. What I meant when I say the other PR will not affect security, is that the Prover changes do not break the system’s constraints—only the circuits and the Solidity verifier do. As for the Rust verifier, it is only used for testing.

The enabling PR is much longer than this one. I separated this PR to ensure it receives a careful review, but in the future I will consider merging them together to provide better context for the reviewers.